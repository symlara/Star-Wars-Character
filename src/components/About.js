import React from 'react';
import headshot from '../assets/images/Leia_Organa.png';

function About() {
    return (
        <article class="pa3 pa5-ns" id="about-me">
            <h1 class="f2">About Me</h1>
            <h2>Leia Organa</h2>
            <img src={headshot} alt="leia"/>
            
        </article>
            
       
    );
}

export default About;