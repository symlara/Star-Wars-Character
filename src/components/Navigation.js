//// Navigation Title must highlight when you're on the page.
// Tabs are clickable
import React from 'react';

function Navigation(props) {
    const tabs = ["About", "Films", "Starships" ];
    return (
        <div>
            <nav>
            <ul className="nav nav-tabs">
        {tabs.map((tab) => (
          <li id="nav-items"
            className={
              props.currentPage === tab ? "nav-item is-active" : "nav-item"
            } 
            key={tab}
          >
            <a 
              href={"#" + tab.toLowerCase()}
              // Whenever a tab is clicked on,
              // the current page is set through the handlePageChange props.
              onClick={() => props.handlePageChange(tab)}
              className={
                props.currentPage === tab ? "nav-link active" : "nav-link"
              }
            >
              {tab}
            </a>
          </li>
        ))}
      </ul>
      </nav>
        </div>
    );
}

export default Navigation;