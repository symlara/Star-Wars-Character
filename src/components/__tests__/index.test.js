// about tests

//We need react to enable the components to function properly so we can test them.
import React from 'react';
// "render" the component.
import { render, cleanup } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';
//we'll import the component we will be testing,
import About from '../About';
import Navigation from '../Navigation';
//This will ensure that after each test, we won't have any leftover memory data that could give us false results.
afterEach(cleanup)

describe('About component', () => {
    // renders About test
    //first test
    it('renders', () => {
        render(<About />);
    });

    // second test
    it('matches snapshot DOM node structure', () => {
        // render About
        const  { asFragment } = render(<About />);
        expect(asFragment()).toMatchSnapshot();
    })
})




const categories = [
    { name: 'portraits', description: 'Portraits of people in my life' }
  ]
  const mockCurrentCategory = jest.fn();
  const mockSetCurrentCategory = jest.fn();
  const mockContactSelected = jest.fn();
  const mockSetContactSelected = jest.fn();

afterEach(cleanup);

describe('Navigation component', () => {
    // baseline test
    it('renders', () => {
      render(<Navigation
        categories={categories}
        setCurrentCategory={mockSetCurrentCategory}
        currentCategory={mockCurrentCategory}
        contactSelected={mockContactSelected}
        setContactSelected={mockSetContactSelected}
      />);
    });

    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Navigation />);
      
        expect(asFragment()).toMatchSnapshot();
      });
})

// emoji test
describe('emoji is visible', () => {
    it('inserts emoji into the h2', () => {
        // arrange
        const {getByLabelText} = render(<Navigation
            categories={categories}
            mockSetCurrentCategory={mockCurrentCategory}
            currentCategory={mockCurrentCategory} 
            contactSelected={mockContactSelected}
            setContactSelected={mockSetContactSelected}
            />);
            
        //assert
        //Here, we have used a custom matcher to compare the expected value to the one received by our query.
        expect(getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
    });
})

// links are visible
describe('links are visible', () => {
    it('inserts text into the links', () => {
      const { getByTestId } = render(<Navigation
        categories={categories}
        mockSetCurrentCategory={mockCurrentCategory}
        currentCategory={mockCurrentCategory} 
        contactSelected={mockContactSelected}
        setContactSelected={mockSetContactSelected}
        />);
      expect(getByTestId('link')).toHaveTextContent('Oh Snap!');
      expect(getByTestId('about')).toHaveTextContent('About me');
    });
  })

