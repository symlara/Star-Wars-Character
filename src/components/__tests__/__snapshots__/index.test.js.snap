// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      id="about-han"
    >
      <h1>
        About Me
      </h1>
      <img
        alt="han-solo"
        src="han-solo.png"
      />
      <div>
        <p>
          <strong>
            Character Height:
          </strong>
          180
        </p>
        <p>
          <strong>
            Character Weight: 
          </strong>
          80
        </p>
        <p>
          <strong>
            Character Hair Color:
          </strong>
          brown
        </p>
        <p>
          <strong>
            Character Birth date:
          </strong>
          29BBY
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <ul
        class="nav nav-tabs"
      >
        <li
          class="nav-item"
          id="nav-items"
        >
          <a
            class="nav-link"
            href="#about"
          >
            About
          </a>
        </li>
        <li
          class="nav-item"
          id="nav-items"
        >
          <a
            class="nav-link"
            href="#films"
          >
            Films
          </a>
        </li>
        <li
          class="nav-item"
          id="nav-items"
        >
          <a
            class="nav-link"
            href="#starships"
          >
            Starships
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
